### PR TITLE
better code for frame polling callback

### PR DESCRIPTION
**Pull request by huitema**
Implements a set of code changes suggested by @igorlord when reviewing PR #465.

### CODE REVIEW

File: picoquic/frames.c [hunk 1]
```diff
@@ -914,75 +914,93 @@ uint8_t* picoquic_decode_stream_frame(picoquic_cnx_t* cnx, uint8_t* bytes, const
...
                  /* Check parity */
-                if (IS_CLIENT_STREAM_ID(stream->stream_id) == cnx->client_mode) {
-                    if (stream->stream_id <= cnx->max_stream_id_bidir_remote) {
-                        break;
-                    }
-                }
-                else {
+                if (IS_CLIENT_STREAM_ID(stream->stream_id) != cnx->client_mode ||
+                    (IS_BIDIR_STREAM_ID(stream->stream_id) && stream->stream_id <= cnx->max_stream_id_bidir_remote) ||
+                    (!IS_BIDIR_STREAM_ID(stream->stream_id) && stream->stream_id <= cnx->max_stream_id_unidir_remote)) {
```
Comment (igorlord) [inline picoquic/frames.c]: You will probably hate my style, but just for the record, I would had written this as:
```
stream->stream_id <= (IS_BIDIR_STREAM_ID(stream->stream_id) ? cnx->max_stream_id_bidir_remote : cnx->max_stream_id_unidir_remote)
```

File: picoquic/sender.c [hunk 2]
```diff
@@ -84,7 +84,7 @@ int picoquic_mark_active_stream(picoquic_cnx_t* cnx,
     if (ret == 0) {
         if (is_active) {
             /* Verify that no data is queued on the stream */
```
Comment (igorlord) [inline picoquic/sender.c]: Fix the comment.

**Comment by huitema** [comment]
Also added the code to split retransmitted frames if the new send mtu is smaller than the previous one. This addresses issue #473.

File: picoquic/frames.c (hunk 1 above)
Comment (huitema) [reply picoquic/frames.c]: Yes, I could save a couple lines. I assume that the compiler will optimize all that, so i went for the easy-to-read form, but having a single line is not that bad.

**Pull request merged**